Lucidtech/Las/Core/Credentials.cs

```diff
@@ -5,67 +5,162 @@
 using IniParser;
 using IniParser.Model;
 
+using RestSharp;
+using RestSharp.Authenticators;
+using Newtonsoft.Json;
+
+using Lucidtech.Las.Utils;
+
 namespace Lucidtech.Las.Core
 {
     /// <summary>
     /// Used to fetch and store credentials. One of 3 conditions must be met to successfully create credentials.
-    /// 1. The path to the file where the credentials are stored is provided
-    /// 2. AccessKeyId, SecretAccessKey and ApiKey are provided
+    /// 1. ClientId, ClientSecret, ApiKey, AuthEndpoint and ApiEndpoint are provided
+    /// 2. The path to the file where the credentials are stored is provided
     /// 3. Credentials are located in default path ~/.lucidtech/credentials.cfg
     /// 
     /// Get credentials by contacting hello@lucidtech.ai
     /// 
     /// </summary>
-    public class Credentials
+    public class AmazonCredentials
```

Comment:
Would be cool to have a possibility to override the credentials with environment variables, could you add this?

Reply:
Should this be the default behaviour?